backend/dataline/services/llm_flow/utils.py

```diff
@@ -0,0 +1,140 @@
+from typing import Any, Iterable, Self, cast
+from langchain_community.utilities.sql_database import SQLDatabase
+from sqlalchemy import URL, Engine, MetaData, create_engine, inspect, text
+from sqlalchemy.engine import CursorResult
+from sqlalchemy.schema import CreateTable
+from sqlalchemy.types import NullType
+
+
+def get_mssql_table_names(engine: Engine):
+    inspector = inspect(engine)
+    schemas = inspector.get_schema_names()
+    db_tables: list[str] = []
+    for schema in schemas:
+        for table_name in inspector.get_table_names(schema=schema):
+            db_tables.append(f"{schema}.{table_name}")
+    return db_tables
+
+
+class DatalineSQLDatabase(SQLDatabase):
+
+    def __init__(
+        self,
+        engine: Engine,
+        schema: str | None = None,
+        metadata: MetaData | None = None,
+        ignore_tables: list[str] | None = None,
+        include_tables: list[str] | None = None,
+        sample_rows_in_table_info: int = 3,
+        indexes_in_table_info: bool = False,
+        custom_table_info: dict | None = None,
+        view_support: bool = True,  # default False in base class
+        max_string_length: int = 300,
+        lazy_table_reflection: bool = False,
+    ):
+        if engine.dialect.name == "mssql":
+            lazy_table_reflection = True  # so that we do our own reflection with multiple schemas
+        super().__init__(
+            engine,
+            schema,
+            metadata,
+            ignore_tables,
+            include_tables,
+            sample_rows_in_table_info,
+            indexes_in_table_info,
+            custom_table_info,
+            view_support,
+            max_string_length,
+            lazy_table_reflection,
+        )
+        if self.dialect == "mssql":
+            inspector = inspect(self._engine)
```

Comment:
More docs here pls - don't wanna have to read the code to understand it 😄 Gen-Z programmer sorry

Reply:
Nah you're right